R/comment_linters.R

```diff
@@ -27,37 +27,42 @@ ops <- list(
 #' blocks
 #' @export
 commented_code_linter <- function(source_file) {
-  res <- re_matches(source_file$file_lines,
-                    rex(some_of("#"), any_spaces,
-                        capture(name = "code",
-                          # except("'"),
-                          anything,
-                          or(some_of("{}[]"), # code-like parentheses
-                            or(ops), # any operator
-                            group(graphs, "(", anything, ")"), # a function call
-                            group("!", alphas) # a negation
-                            ),
-                          anything
-                        )
-                    ),
-                    global = FALSE, locations = TRUE)
+  if (is.null(source_file$full_xml_parsed_content)) return(list())
+  all_comment_nodes <- xml2::xml_find_all(source_file$full_xml_parsed_content, "//COMMENT")
+  all_comments <- xml2::xml_text(all_comment_nodes)
+  code_candidates <- re_matches(
+    all_comments,
+    rex(some_of("#"), any_spaces,
```

Comment:
this part isn't needed anymore right? ditto for `anything` in the capture?

Reply:
I think we need to anchor on the (first) `#` of the comment text, which will always be there in order to get the `column_offset` correct, but I'm not completely sure.

It depends on how the start and end locations of capture groups are returned by `rex`. If they are relative to the input string, we can probably prune the regex to only extract the code candidate to be analyzed. Note that we will still need to somehow anchor it to just after the comment start token, or it will match the entire string including the `#` which would always return a parseable (comment-only) code.

The COMMENT token will contain the `#` character starting the comment, so the refactoring as is is more or less sure to deliver the same candidate code as before if no `#` is present in actual code. WDYT?

Reply:
Sure, agree it's safer not to try & rock the boat during a refactor.